src/Disks/IO/ThreadPoolRemoteFSReader.cpp

```diff
@@ -66,6 +68,15 @@ ThreadPoolRemoteFSReader::ThreadPoolRemoteFSReader(size_t pool_size, size_t queu
 
 std::future<IAsynchronousReader::Result> ThreadPoolRemoteFSReader::submit(Request request)
 {
+    auto * fd = assert_cast<RemoteFSFileDescriptor *>(request.descriptor.get());
+    if (fd->getReader().contentIsCached())
```

Comment:
But `file_offset_of_buffer_end` in `CachedOnDiskReadBufferFromFile` could be different at this point from what we have in `request.offset`, so here we can get incorrect result of `contentIsCached`. May be at least check that if `!cached_buffer_initialized || current_file_segment_range_in_cache_buffer.contains(request.offset)` => we can safely check `contentIsCached`? Btw also we know that `seek` does not do anything except for resetting the buffer pointer and creating a new buffer, then may be we could do the seek here and have a correct offset position in cached buffer which will allow us to check `contentIsCached` even if offsets mismatch? Or it does not worth it?
Check `current_file_segment_range_in_cache_buffer.contains(request.offset)` could also be extended to `cached_buffer->file_segments.contains(request.offset)` (there is no such method in code, but it is pseudocode)

Reply:
also it seems `isContentCached` easier readable than `contentIsCached`? I am just more used to methods starting with `is*` instead of `is` in the middle of method name as there are many such methods in our code which start with `is`. But this is just a suggestion :) I do not mind if it will be `contentIsCached`.

Reply:
Also continuing my first comment, it could be convenient to have `contentIsCached` getting and argument `contentIsCached(size_t offset)` which will allow to check cache on specific offset (and if this offset goes beyond currently hold list of file segments (in FileSegmentsHolder inside CachedOnDiskReadBufferFromFile) in cached buffer then it should be ok to return false instead of checking with the `cache` object)

Reply:
Though this will not be needed if we do `seek && initialise` as I wrote in first comment, but not sure if it is ok.